docs/user/git_workflow.md

```diff
@@ -0,0 +1,78 @@
+
+Note that, in some cases, you may touch a file with code that has not been updated to adhere to garage's latest commit hooks. Garage's policy is that all pre-commit hooks should pass, even if errors are due to existing code not committed by you. It is therefore your responsibility to update the touched files and make them compliant. Exceptions to this rule are only made for large PRs that touch many files.
+
+## Rebase - Don't Merge
+
+If you have implemented a feature, its likely that the changes you introduced were not created on top of the most recent commit on the garage master branch (ie. commits were made to master that you don't have locally). To update your feature branch to include the latest commits made to `rlworkgroup/master`, you should rebase your commits on top of them. This will also avoid conflicts between your commits and the commits on the remote repository.
+
+If you're unfamiliar with rebasing, [this](https://www.atlassian.com/git/tutorials/rewriting-history/git-rebase#:~:text=Rebase%20is%20one%20of%20two,has%20powerful%20history%20rewriting%20features.) tutorial explains it in detail. Note that you should *only* rebase when creating a PR, never`git merge`. Garage maintains a linear commit history and does not allow merge commits.
+
+The [CONTRIBUTING.md](https://github.com/rlworkgroup/garage/blob/master/CONTRIBUTING.md#git-recipes) includes instructions on how to rebase, but they are reiterated here for further clarity:
+
+ **1. Make sure you have the rlworkgroup repo as a remote:**
+
+```sh
+git remote add rlworkgroup git@github.com:rlworkgroup/garage.git
+```
+
+**2. Update your local master branch to include the latest changes to the
+upstream repo**
+
+Note that doing this will overrwrite any changes you've made to your master branch locally. Make sure you move any changes you've made to a new branch first, then:
+
+```sh
+git checkout master
+git fetch rlworkgroup
+git reset --hard rlworkgroup/master # update the local master branch
+git push -f origin master # optional, this updates the master branch on your fork
+```
+
+**3. Rebase your changes onto master:**
+
+```sh
+git checkout your-feature-branch
+git rebase master # you may have to resolve conflicts here. Follow git's instructions on how to do this.
+```
+
+You can verify that the rebase was successful by doing another rebase with `git rebase master`. You should see a message telling you the current branch is up to date.
+
+## Create a PR (Optional)
```

Comment:
maybe add some later notes about how to backport (it's just a PR directed at the release branches)

in the same section you can mention the role of release branches and tags.

Reply:
I included a section on backporting and referenced the preparing a PR doc for details on how to backport/use tags.